pkg/server/preprocessor/auth.go

```diff
@@ -113,9 +114,18 @@ func (p *UserAuthenticator) Preprocess(payload *router.Payload, response *router
 	// For master access key, it is possible to impersonate any user of
 	// the caller's choosing.
 	if payload.HasMasterKey() {
-		if userID, ok := payload.Data["_user_id"].(string); ok {
-			payload.UserInfoID = userID
-			payload.Context = context.WithValue(payload.Context, router.UserIDContextKey, userID)
+		userID, ok := payload.Data["_user_id"].(string)
+		if !ok {
+			userID = "_god"
+		}
+		payload.UserInfoID = userID
+		payload.Context = context.WithValue(payload.Context, router.UserIDContextKey, userID)
+		// Try to create the implicitly user if it does not exist
+		info := skydb.UserInfo{
+			ID: userID,
+		}
+		if err := payload.DBConn.CreateUser(&info); err != nil && err != skydb.ErrUserDuplicated {
```

Comment:
Oh sorry, I just looked at the preprocessor folder and decided `auth.go` seems to be the most relevant place to add function, without realising there is inject user. After moving things to inject user, issues in your other comments are gone!